File: src/main/java/fr/marstech/mtlinkspray/controller/ViewShortenerController.java
package fr.marstech.mtlinkspray.controller;

import fr.marstech.mtlinkspray.controller.api.ApiShortenerController;
import fr.marstech.mtlinkspray.enums.ViewNameEnum;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.validation.constraints.NotBlank;
import org.springframework.stereotype.Controller;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.servlet.ModelAndView;

import static fr.marstech.mtlinkspray.enums.ViewNameEnum.SHORTENER;

@Validated
@Controller
public class ViewShortenerController implements ThymeleafViewControllerInterface {

  private static final ViewNameEnum viewNameEnum = SHORTENER;

  final ApiShortenerController apiShortenerController;

  public ViewShortenerController(ApiShortenerController apiShortenerController) {
    this.apiShortenerController = apiShortenerController;
  }

  @Override
  public ModelAndView getModelAndView() {
    return getModelAndView(viewNameEnum);
  }

  @GetMapping("/shortener")
  public ModelAndView getView() {
    return getModelAndView();
  }

  @PostMapping("/shortener")
  public ModelAndView postView(
      @RequestParam(name = "input-link") @NotBlank String inputLink,
      HttpServletRequest httpServletRequest) {
    return getModelAndView()
        .addObject("inputLink", inputLink)
        .addObject(
            "shortenedLink", apiShortenerController.getShort(inputLink, httpServletRequest));
  }
}

File: src/main/java/fr/marstech/mtlinkspray/controller/view/ViewShortenerController.kt
package fr.marstech.mtlinkspray.controller.view

import fr.marstech.mtlinkspray.controller.api.ShortenerApiController
import fr.marstech.mtlinkspray.enums.ViewNameEnum.SHORTENER
import jakarta.servlet.http.HttpServletRequest
import jakarta.validation.constraints.NotBlank
import org.springframework.stereotype.Controller
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.servlet.ModelAndView

@Validated
@Controller
@RequestMapping("/shortener")
open class ViewShortenerController(val shortenerApiController: ShortenerApiController) :
    ThymeleafViewControllerInterface {

    override fun getModelAndView() = getModelAndView(SHORTENER)

    @GetMapping
    fun getShortenerPage(): ModelAndView = getModelAndView()

    @PostMapping
    fun createShortenedLink(
        @RequestParam(name = "input-link") inputLink: @NotBlank String?,
        httpServletRequest: HttpServletRequest?
    ): ModelAndView = getModelAndView()
        .addObject("inputLink", inputLink)
        .addObject("shortenedLink", shortenerApiController.getShort(inputLink, httpServletRequest))
}
